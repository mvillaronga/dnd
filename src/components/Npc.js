import React from 'react'
import PropTypes from 'prop-types'

const Npc = ({ onClick, name }) => (
  <li
    onClick={onClick}
  >
    {name}
  </li>
)

Npc.propTypes = {
  onClick: PropTypes.func.isRequired,
  name: PropTypes.string.isRequired
}

export default Npc