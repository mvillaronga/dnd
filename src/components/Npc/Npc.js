import React from 'react'
import PropTypes from 'prop-types'

const Npc = ({ onClick, name, selected }) => (
  <li
    style={{
      color: selected ? 'red' : 'blue'
    }}
    onClick={onClick}
  >
    {name}
  </li>
)

Npc.propTypes = {
  onClick: PropTypes.func.isRequired,
  name: PropTypes.string.isRequired
}

export default Npc