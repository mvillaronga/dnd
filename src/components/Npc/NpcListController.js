import React from 'react'
import { useDispatch, useSelector } from 'react-redux'

import {toggleNPC} from '../../redux/actions'

import NpcList from './NpcList'

const NpcListController = () => {
  const dispatch = useDispatch()

  const props = {
    npcs: useSelector(state => state.npcs),

    onNpcClick: (id) => {
      dispatch(toggleNPC(id))
    }
  }

  return <NpcList {...props} />
}

export default NpcListController