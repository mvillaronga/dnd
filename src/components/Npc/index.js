import React from 'react'
import { useDispatch, useSelector } from 'react-redux'

import { toggleNpc } from '../../redux/npcs'

import NpcList from './NpcList'

const NpcListController = () => {
  const dispatch = useDispatch()

  const props = {
    npcs: useSelector(state => state.npcs),

    onNpcClick: (id) => {
      dispatch(toggleNpc(id))
    }
  }

  return <NpcList {...props} />
}

export default NpcListController