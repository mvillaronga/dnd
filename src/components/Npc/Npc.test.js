import React from 'react';
import Npc from './Npc';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

configure({adapter: new Adapter()});
describe('Npc component tests', () => {
  it('Default Spell Card Rendering', () => {
    const component = renderer.create(
      <Npc name="Bobo" onClick={jest.fn()} selected="false" />,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('On Click handler', () => {
    const onClick = jest.fn()

    const component = shallow(
      <Npc name="Bobo" onClick={onClick} selected="false" />,
    );

    component.find("li").first().simulate('click')

    expect(onClick).toHaveBeenCalled()
  })
})
