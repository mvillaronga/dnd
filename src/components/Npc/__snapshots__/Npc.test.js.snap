// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Npc component tests Default Spell Card Rendering 1`] = `
<li
  onClick={[MockFunction]}
  style={
    Object {
      "color": "red",
    }
  }
>
  Bobo
</li>
`;
